Eating from floor may cause disease

MTandi opened the pull request:
![image](https://github.com/tgstation/tgstation/assets/3625094/a60ac166-5772-4ee1-aa08-4f82828033e7)

## About The Pull Request

There is a 10% chance of getting one of 3 new diseases when you eat dirty things.

Things become dirty when left on the floor for [more than 5 seconds](https://en.wikipedia.org/wiki/Five-second_rule).

But you can wash (with any method you know from spraying water to cleaning with soap) or cook them later to avoid this.

![image](https://github.com/tgstation/tgstation/assets/3625094/0f78ef11-1737-4c9c-aecd-072dd95ef013)

Packaged, bowled, canned food (any food that spawns package as trash afterwards) is protected from this effect.

Makes crafted food spawn on nearby tables when the hands are full. Except the one behind you.

![8wSPp0hsx1](https://github.com/tgstation/tgstation/assets/3625094/6a51ebce-5626-4aac-9e59-cc4eab46a95a)

#### New diseases:

40% chance:
![image](https://github.com/tgstation/tgstation/assets/3625094/a0b72459-cc10-47e1-99db-b11013eaa61b)

40% chance (Vomiting is of special type that does not stun):
![image](https://github.com/tgstation/tgstation/assets/3625094/ce8e254c-7a65-49ed-bfb2-68652a624aed)

20% chance:
![image](https://github.com/tgstation/tgstation/assets/3625094/b792ae5e-8a99-4271-93d1-bcc172292049)

## Why It's Good For The Game

Things that are left on the floor for too long intentionally are trash that should be disposed by janitor. If you make a meal or prepare a medication, it makes sense that you should keep your product sanitized.

Things that are dropped unintentionally are supposed to be picked up quickly. "Oops I dropped this pie, need to pick it up quickly before the germs spread". 5 seconds are enough for this. If you didn't manage you will be like "Oh dammit, now I need to wash this pie in a sink".

Now players will consider to not just throw items meant for eating onto the floor neglecting the fact that it looks odd. If they still ignore it, people who consume the items will receive a harmless but annoying disease.

In general this PR aims to force some IC gameplay onto Medics, Chefs and Botanists so that they care a bit more about things they make for other players.

The items have a warning message saying that they are dirty and dangerous, so the consumers have a way to detect dirty items and an option to wash them with soap/rag/sink/shower/fire extinguisher to remove the harmful part from the edible item.

So to avoid this, players just need to examine an item before eating it.

Botanists can spray a pile of fruits from a hose for the same effect, and washed items that stay on floor dont regain germs until moved to another tile.

Food that converts into another item during cooking (like meat slab turning into steak) or crafting, will not retain the infection. This kinda simulates the sanitizing during cooking.

Medics can use elevated structures (e.g. conveyor belt) to avoid getting their pills dirty during creation in plumbing. Or they can wash the pills they want to distribute in the shower before packaging them into pill bottles or a bag.

## Changelog

:cl:
add: Food and pills have a 10% chance to infect with one of three new diseases on consumption when left for more than 5 seconds on the floor. You can wash it to avoid disease. ChemMaster and Pill Press are added to the list of elevated structures (Considered as tables for pills). Made harvest spawn on top of hydrotrays to stay protected from germs.
add: Added three new advanced diseases: Gastritium, Carpellosis, Nebula Nausea with static cures obtained by digesting dirty food.
fix: Food no longer decomposes on Hydrotrays, Grilles, Bonfires and all dense kitchen machinery
code: Decomposition now uses `germ_sensitive` component and follows 5 second rule too.
qol: Crafted food items spawns on nearby tables (except the one behind you) instead of dropping on floor when hands are full.
/:cl:

MTandi committed:
dont eat shit from floor

MTandi committed:
#undef

JohnFulpWillard commented:
All chef food would be infected this way, since crafting food en-masse ends up with a lot on the floor in short periods.
Maybe it should only work if the floor under it is actually dirty? Or anything that Chefs can use, like their serving tray, can maybe get food automatically put in there instead of to the floor.

As it stands currently I like the idea (I've wanted to do something similar but it wasn't well received), but we gotta do something about the Chef problem I feel.

MTandi commented:
> All chef food would be infected this way, since crafting food en-masse ends up with a lot on the floor in short periods. Maybe it should only work if the floor under it is actually dirty? Or anything that Chefs can use, like their serving tray, can maybe get food automatically put in there instead of to the floor.
> 
> As it stands currently I like the idea (I've wanted to do something similar but it wasn't well received), but we gotta do something about the Chef problem I feel.

The diseases are not guaranteed, harmless (usually cured by basic reagents and at maximum only leads to sniffing, sneezing and annoying chat messages.

ExcessiveUseOfCobblestone commented:
botany plants are also problematic if you dont use a bag but then again ever since they could decomp the plants should have been spawning on the tray and the tray be considered a safespot

MTandi commented:
> botany plants are also problematic if you dont use a bag but then again ever since they could decomp the plants should have been spawning on the tray and the tray be considered a safespot

I'm going to add washing mechanic.

Cheshify commented:
It's like ants but even more punishing

Helg2 commented:
> 5 seconds is VERY short

5 seconds + Time dilation

Helg2 commented:
Blacklist for food in containers and bowls perhaps?

Farquaar commented:
I like the idea, but it should probably be more than 5 seconds. Like, a minute would make more sense.

Paxilmaniac commented:
Five second rule.

MTandi committed:
wash your food

MTandi commented on code/datums/components/food/bacteria_carrier.dm:
```diff
@@ -0,0 +1,100 @@
+	for(var/atom/movable/content as anything in open_turf.contents)
+		if(GLOB.typecache_elevated_structures[content.type])
+			remove_timer()
+			return
```
And pills should be considered as preserved food?

MrMelbert replied:
So decomposition can be expanded to encompass both behaviors, perhaps via callbacks?

Or the "is on table" portion of decomposition can be separated out (perhaps into an element) that fire ssignals when an item is safe / unsafe?

There's just a bit of overhead to the current method of copy pasting decomposition behavior i'm not too plussed about

MTandi replied:
>Or the "is on table" portion of decomposition can be separated out (perhaps into an element) that fire ssignals when an item is safe / unsafe?

Thanks for suggestion. I'll try to refactor it into a component like "germ_exposure" that will fire signals that are used by both decomposition and germ_carrier. It will probably lead to decomposition being affected by dirtyness of the tile or whether people walk on it, but this makes sense.

DATA-xPUNGED commented:
mass crafting anything as chef will inevitably leave stuff on the ground

MTandi commented:
> mass crafting anything as chef will inevitably leave stuff on the ground

Only when your hands are not empty.

MTandi committed:
Merge branch 'master' into floor_food

MTandi committed:
exclude bowled, canned, snacks with package

moocowswag commented:
What about conveyors

Dorsisdwarf commented:
> > mass crafting anything as chef will inevitably leave stuff on the ground
> 
> Only when your hands are not empty.

When you craft multiple foods only the first goes in your hands. The rest get dumped on the floor. 

This is basically griefing all cooks and botanists and anyone who doesn't eat out of a vending machine for the sake of a very mediocre joke.

moocowswag commented:
Also does this impact everything thats edible? Including stuff that you spray fried.

lessthnthree commented:
admin spawned food is gonna hit the floor, is that exempt

lessthnthree commented on code/datums/components/food/germ_carrier.dm:
```diff
@@ -0,0 +1,100 @@
+
+#undef FIVE_SECOND_RULE
+#undef MAX_DISEASE_SYMPTOMS
+#undef MAX_DISEASE_STRENTH
```
STRENGTH

lessthnthree commented on code/datums/components/food/germ_carrier.dm:
```diff
@@ -0,0 +1,100 @@
+
+/datum/component/germ_carrier/proc/infect_parent()
+	infective = TRUE
+	var/datum/disease/advance/random/random_disease = new(max_symptoms = rand(MAX_DISEASE_SYMPTOMS), max_level = rand(MAX_DISEASE_STRENTH))
```
DISEASE_SPREAD_CONTACT_FLUIDS / DISEASE_SPREAD_CONTACT_SKIN / DISEASE_SPREAD_AIRBORNE
up to you

lessthnthree replied:
See "Can you not then just drop a piece of food on the floor intentionally, eat it, and then start a disease outbreak. Repeat at will."

MTandi replied:
It will be overwritten with stages and they can already do this by walking on rotting gibs, without this PR

Mothblocks replied:
i think i agree that excluding admin stuff from this doesnt really make sense

MTandi replied:
Irrelevant due to custom diseases

MTandi committed:
typo

lessthnthree commented on code/datums/components/food/germ_carrier.dm:
```diff
@@ -0,0 +1,100 @@
+		if(GLOB.typecache_elevated_structures[content.type])
+			remove_timer()
+			return
+
```
```
if(flags_1 & ADMIN_SPAWNED_1)
	remove_timer()
		return
```

MTandi replied:
Admins should spawn food on tables.

lessthnthree replied:
Why? If they want it to be infective they can add it. When setting up events and such sometimes you don't have the time to comb through all you spawn to see if it's infected or not.

Mothblocks replied:
i think i agree that making this exclude admin stuff doesnt make much sense

lessthnthree commented on code/datums/components/food/germ_carrier.dm:
```diff
@@ -0,0 +1,100 @@
+/datum/component/germ_carrier/proc/examine(datum/source, mob/user, list/examine_list)
+	SIGNAL_HANDLER
+	if(infective)
+		examine_list += span_notice("[parent] looks dirty and not safe to consume.")
```
examine_list += span_warning("[parent] looks dirty and not safe to consume.")

LemonInTheDark commented:
Crafted food should place on a table if one is near you (starting with your facing direction and working backwards), and plants should dump on the tray.
Foodcrafting dumping on the floor is a cringe thing regardless of this, this change only serves to highlight its bad vibes.

bluetshirtguy commented:
Does chef really need pr's that stifle freedom and ways to play?

the-orange-cow commented:
This seems overly punishing.  As various people have mentioned, a lot of crafting/botany automatically end up on the floor.  This would create a lot of tedium just to make sure the food product doesn't become disease spreading.  I don't really see how this benefits the game, considering we already have decomposition.

Mothblocks reviewed (changes requested):
> Buffs tables.

this is a nothing explanation for why its good for the game. please write one

Mothblocks commented on code/datums/components/food/germ_carrier.dm:
```diff
@@ -0,0 +1,100 @@
+// Don't eat off the floor or hold parent object with dirty hands, you'll get sick
+
+/// Time needed for bacteria to infect the parent object
+#define FIVE_SECOND_RULE (5 SECONDS)
```
actual variable name, especially since i dont think this should be just 5 seconds anyway

Mothblocks commented on code/datums/components/food/germ_carrier.dm:
```diff
@@ -0,0 +1,100 @@
+/// Max strength of the random disease
+#define MAX_DISEASE_STRENGTH 3
+
+/datum/component/germ_carrier
```
document this

Mothblocks commented on code/game/objects/items.dm:
```diff
@@ -1311,6 +1311,12 @@
 	. = ..()
 
+	// Wash germs off dirty things
+	var/datum/component/germ_carrier/germs = GetComponent(/datum/component/germ_carrier)
+	if(germs?.infective)
+		germs.infective = FALSE
+		qdel(GetComponent(/datum/component/infective))
+
```
don't do this, make germ_carrier register a signal to wash

GuillaumePrata commented:
This will end the annoying 50+ pile of pills at the entrance of medbay.
I also wanted to PR something like this, it is really silly that the most efficient way to get people to use vaccines is to dump pills on the floor, and if you use the non spam way of using the fridge, people will ignore it and break into the pharmacy instead...

ChungusGamer666 commented:
good pr but would be better if floor food got stink lines

ATH1909 commented:
cure low res deathviruses with this one neat trick!

KittyNoodle commented:
it really should only effect the person who ate the food. i don't like it punishing some rando for some assistant deciding to drop food on the floor for 6 seconds before eating it

Striders13 commented:
it should give whoever eats it tummy ache (so assistants can be brave about it) instead of transmitive disease

breadcrombs commented:
We already have ants that "buffs tables" this just adds another unnecessary feature to be not putting foods on floor. Not only is the timer too fast but the virus could be actually useful to possibly remove sentient and other harmful viruses.  It feels like a really unneeded and out of place feature since we already have ants. It only moves the pill spam to the tables, which still gives the same tile spam lag.

KittyNoodle commented:
if you make it a 5 minute timer that instead gives you a static non-transmittable disease it would be better.

in the event that someone dumps 200 pills onto the floor of medical lobby for whatever reason, thats 200 SEPERATE viruses with SEPRATE cures that all need SEPERATE vaccines. I am convinced this pr is a psyop to get virology removed

Livrah commented:
Everyone who played more then 2 rounds: "remove diseases"
Coder: "got you"

MTandi commented:
> if you make it a 5 minute timer that instead gives you a static non-transmittable disease it would be better.
> 
> in the event that someone dumps 200 pills onto the floor of medical lobby for whatever reason, thats 200 SEPERATE viruses with SEPRATE cures that all need SEPERATE vaccines. I am convinced this pr is a psyop to get virology removed

You may be amazed, but killing a few monkeys in a hallway and letting the gibs objects rot already does what you described, but may lead to a more harmful disease than the ones added in this PR.

>Everyone who played more then 2 rounds: "remove diseases"
>Coder: "got you"

Not everyone play the game to make fun gimmicks and entertain their audience. I remember you complaining about diseases because of their lethality and unknown source. This PR adds HARMLESS diseases that may even overwrite the bat virus of yours at the cost of messages that you're itching. And you can cure it just by drinking some iron or washing the pill before eating it.

Exester509 commented:
Except that people can leave food or pills on the floor for legitimate reasons, and this would be a severe punishment for that. You make a pill factory for reagents that are helping keep medbay going, the doctor grabs one and feeds it to a patient in a hurry. Congratulations, you have now infected someone with a virus.
And god forbid if one of these doctors even forgets about this PR, it's going to go from Medical Doctor to Virologist in the blink of an eye.

But hey, it's not like we have mold, or ants, or anything like that, right?

Exester509 commented:
Also, if the diseases are absolutely harmless, what's the point? This will be absolutely meaningless and perfectly ignorable 100% of the time. If you implemented something like this with actual effects and over some time, so that people at least check what they're eating, I wouldn't be against it, but a completely harmless disease won't affect the gameplay at all, and a harmful one would need more time before the food becomes dirty, which is also the main proposed change

MTandi commented:
> Except that people can leave food or pills on the floor for legitimate reasons, and this would be a severe punishment for that. 

What are these reasons?

> You make a pill factory for reagents that are helping keep medbay going, the doctor grabs one and feeds it to a patient in a hurry. Congratulations, you have now infected someone with a virus. And god forbid if one of these doctors even forgets about this PR, it's going to go from Medical Doctor to Virologist in the blink of an eye.

There are ways to keep pills clean in pill factory. The disease will be harmless and the reason the patient came for this pill will be resolved regardless. A good doctor should not grab a pill from a floor, or at least spray the pill pile with space cleaner before feeding it to people. They will learn eventually.

>Also, if the diseases are absolutely harmless, what's the point?

Please red the opening post. I've added the proper section for "Why is it good for the game" addressing these points.

MTandi commented:
> We already have ants that "buffs tables" this just adds another unnecessary feature to be not putting foods on floor. Not only is the timer too fast but the virus could be actually useful to possibly remove sentient and other harmful viruses. It feels like a really unneeded and out of place feature since we already have ants. It only moves the pill spam to the tables, which still gives the same tile spam lag.

You just said why this PR is good - now as a sentient disease or traitor viro you will need to think about your stats better to not become a victim to harmless diseases, which means that they will have less options to improve harmful effects of their symptoms. So it's an indirect nerf to custom diseases.

MTandi commented:
> good pr but would be better if floor food got stink lines

You mean it should be swabbable for cytology things?

Exester509 commented:
> > Except that people can leave food or pills on the floor for legitimate reasons, and this would be a severe punishment for that.
> 
> What are these reasons?
> 
> > You make a pill factory for reagents that are helping keep medbay going, the doctor grabs one and feeds it to a patient in a hurry. Congratulations, you have now infected someone with a virus. And god forbid if one of these doctors even forgets about this PR, it's going to go from Medical Doctor to Virologist in the blink of an eye.
> 
> There are ways to keep pills clean in pill factory. The disease will be harmless and the reason the patient came for this pill will be resolved regardless. A good doctor should not grab a pill from a floor, or at least spray the pill pile with space cleaner before feeding it to people. They will learn eventually.
> 
> > Also, if the diseases are absolutely harmless, what's the point?
> 
> Please red the opening post. I've added the proper section for "Why is it good for the game" addressing these points.

This still doesn't adress how does this make this PR relevant. If the disease is harmless there's no punishment for ignoring it completely and in fact, no reason not to, since getting these diseases will only give you an advantage over low resistance harmful ones.
Also, the "why is this good for the game" isn't an explanation as much as it is your thoughts of what realism should look like in a rather unrealistic game.
I really see no advantage of having this merged, if I'm being honest

MTandi commented:
>Also, the "why is this good for the game" isn't an explanation as much as it is your thoughts of what realism should look like in a rather unrealistic game.

I go with the assumption that logic and realism should be the key to game design in SS13. There are so many things and mechanics that you won't be able to learn them, so if we use our IRL experience, the game will be more intuitive to the players without requiring them to learn the exact mechanics.

If you eat shit from floor, it should lead to bad consequences - that's the information any new player comes with into the game, and the game should use this fact.

Whether you agree with i or not, is another topic. It makes sense why this PR is so controversial and polarizing and it will be up to maintainers what to do in the end. I think that this will make the game better. If they think otherwise, then I'm fine with that opinion if it's justified.

breadcrombs commented:
> > We already have ants that "buffs tables" this just adds another unnecessary feature to be not putting foods on floor. Not only is the timer too fast but the virus could be actually useful to possibly remove sentient and other harmful viruses. It feels like a really unneeded and out of place feature since we already have ants. It only moves the pill spam to the tables, which still gives the same tile spam lag.
> 
> You just said why this PR is good - now as a sentient disease or traitor viro you will need to think about your stats better to not become a victim to harmless diseases, which means that they will have less options to improve harmful effects of their symptoms. So it's an indirect nerf to custom diseases.

Yeah that is why this pr is ACUTALLY bad and not good, it indirectly affects and restricts other gameplay mechanics in unintended ways and gives so many oversights. Having a pr that unintentionally nerfs or buffs other mechanics is the issue.

Not only will it mess up viro balance, it will mess up bulk cooking for chef considering the few tables(excluding the main counter that other players can access) they have in certain maps. This just went from a simple buffed ants pr to a pr that will add so many complex and unneeded rp restrictions to gameplay mechanics. Having another mechanic to discourage eating from the floor is somewhat okay, having it drastically alter gameplay mechanic is not. 

Also using realism and logic is invalid when you can "wash" cooked foods. Does that mean I can cook a piece of meat, drop it on the floor for 30 seconds and then wash it and it will be as if it's a normal piece of cooked meat. What about eating a raw piece of meat and not getting diseases? Also getting random diseases unrelated to mainly bad food diseases is also unrealistic.  Either way, using logic/realism is invalid and lame when you can make multiple arguments for and against it. 

I really like the other QoL updates included with this but I don't agree with the virus part. I suggest it only makes floor foods less tasty and have slight disgust. Having it make food unappetizing will have more direct effect than a harmless virus which will ruin game balance.

Roryl-c commented:
Anecdotally, I've always played chef respecting food (never letting stuff be on the floor) so I'm entirely unbothered by this change. When bulk crafting it's very easy to hit Shift+E between each craft or click your serving tray to avoid overflowing product into the floor at negligible time loss. 

The floor pills method of distributing vaccines (which is by far the most successful - nobody checks chem fridges) does get impacted but that always struck me as odd anyway, so we could probably use a more formal solution to that in general in a future PR, like some kind of signage or a pill dispenser machine.

HoodMilk commented:
Just make it use already existing harmless diseases like the cold or flu, so that virologists don't have to make 15 vaccines a round.

Or alternatively just not have things being on the floors give you diseases since the entire virology system is just cbt and isn't fun.

MTandi committed:
code feedback and refactor

MTandi committed:
pill press considered as a table

ChungusGamer666 commented on code/datums/components/food/decomposition.dm:
```diff
@@ -24,7 +25,7 @@
 
 /datum/component/decomposition/Initialize(mapload, decomp_req_handle, decomp_flags = NONE, decomp_result, ant_attracting = FALSE, custom_time = 0)
-	if(!isobj(parent))
+	if(!isobj(parent) || !parent.GetComponent(/datum/component/germ_sensitive))
```
should use a trait, something like TRAIT_GERM_SENSITIVE, to avoid the GetComponent call

ChungusGamer666 commented on code/modules/reagents/reagent_containers/pill.dm:
```diff
@@ -48,7 +50,8 @@
 	if(icon_state == "pill4" && prob(5)) //you take the red pill - you stay in Wonderland, and I show you how deep the rabbit hole goes
 		addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(to_chat), M, span_notice("[pick(strings(REDPILL_FILE, "redpill_questions"))]")), 50)
-
+	if(apply_type == INGEST)
+		SEND_SIGNAL(src, COMSIG_FOOD_EATEN, eater = M, feeder = user)
```
this looks evil, why not create a new signal instead

Exester509 commented:
> > Also, the "why is this good for the game" isn't an explanation as much as it is your thoughts of what realism should look like in a rather unrealistic game.
> 
> I go with the assumption that logic and realism should be the key to game design in SS13. There are so many things and mechanics that you won't be able to learn them, so if we use our IRL experience, the game will be more intuitive to the players without requiring them to learn the exact mechanics.
> 
> If you eat shit from floor, it should lead to bad consequences - that's the information any new player comes with into the game, and the game should use this fact.
> 
> Whether you agree with it or not is another topic. It makes sense why this PR is so controversial and polarizing and it will be up to maintainers what to do in the end. I think that this will make the game better. If they think otherwise, then I'll comply with that opinion if it's justified.

Thing is a LOT of things aren't logical nor realistic, core game mechanics, too. We have real and fictional gasses at impossible temperatures, very rare chemicals appearing from thin air, a cooking and botany system that doesn't take what it realistically should, and virus that kill within minutes instead of actual days. Nobody is going to complain about realism if you don't have to do every single menial task from real life.
Also, you talk about realism, but you can drop something on the floor for five seconds and not get a DISEASE.

Rex9001 commented:
I think I will echo what I said on the discord thread of a similar project, why does this need to be a disease over just making it a mood debuff or similar? Also this would be "harmless" but its actually quite hard to Cure some of these "harmless" viruses since some of the symptoms (like coughing) can create stuns which close the chemistry menu making it considerably more annoying and time wasting to cure.

MTandi committed:
put on nearby table

MTandi committed:
this is intentional

MTandi committed:
hydrotrays harvest drops on top of hydrotray

MTandi committed:
separate COMSIG_PILL_CONSUMED

MTandi committed:
no germ exposure in zero gravity

MTandi committed:
fixed component logic

MTandi committed:
cleaner

MTandi committed:
harvest unit test fix

MTandi commented:
> Thing is a LOT of things aren't logical nor realistic, core game mechanics, too. We have real and fictional gasses at impossible temperatures, very rare chemicals appearing from thin air, a cooking and botany system that doesn't take what it realistically should, and virus that kill within minutes instead of actual days. Nobody is going to complain about realism if you don't have to do every single menial task from real life. Also, you talk about realism, but you can drop something on the floor for five seconds and not get a DISEASE.

I will make a list of custom diseases for this one so that they will be less transmissive and not so annoying.
You may not get sick if you eat from the floor, the chance of infection is small, and I'm considering reducing it because the roll happens on every bite.

And I see this mechanic as a possibility for chemists and chefs to fuck up at their job, this is what makes the game fun. When engineers fuck up, the station ends up with a hole in the floor and no power. 

>Also this would be "harmless" but its actually quite hard to Cure some of these "harmless" viruses since some of the symptoms (like coughing) can create stuns which close the chemistry menu making it considerably more annoying and time wasting to cure.

I will come up with a list of diseases with set cure and symptoms. Virologist will actually get another activity with this - they can try to get sick from dropped food and prepare vaccines for these deseases beforehand. Chemist can do the same, but prepare these pre-defined medicines beforehand. 

So if someone with food poisoning comes to chemistry, they may already find the needed medicine in a smartfridge if Viro/Chemists are being proactive.

MTandi committed:
more unit test fixes

GPeckman commented:
One consequence of this is that ghetto virology will become much easier. Most of the things used for virology (Pandemic machine, chem dispenser for chems) can be obtained pretty easily with a circuit printer board and some parts. But to actually make a virus you need an infected blood sample to serve as a base. Previously, to get such a blood sample you'd have to break into virology, order it from cargo, or wait to get infected yourself. Or try and use gibs to get yourself infected, I guess. Now there's an even easier way to do so.

I don't know if this is a good consequence or a bad one, but it is a consequence.

MTandi committed:
mood event

Darkshadowrogue commented:
I made a github account solely because I play chef and this offends me greatly. I know you've listed reasons for why you think this change is good for the game, but I don't.

the-orange-cow commented:
> Clicking a "breathe" button every 5 seconds so your character gets enough oxygen is also "realistic", but by no stretch of the imagination would it make the game more fun.

Oh, we already have that mechanic in cursed hearth and corazargh.  It is exactly as 'fun' as you might expect.

MTandi commented:
@Darkshadowrogue I do play chef. I play it like @Roryl-c above, respecting food hygiene for no real reason. This doesn't add any extra overhead in terms of tedium. All you need is just to not drop stuff on the floor deliberately. Many chefs already avoid this for IC reasons. And for others I made a few improvements in scope of this PR:

- Food left on griddle will no longer decompose, so you can leave your eggs and steaks there safely now
- When you craft food and your hands are full, it is automatically placed on nearby tables

Ants are not really punishing because of very long delay. You need the food to really stay on floor for a long time for ants to spawn, so it is barely a punishment.
And I am already making adjustments that should address many concerns in comments above, so please keep in mind the whole picture.

- It will be picked from a list of diseases that will be unique, have static medicine, non-transmissive, so that it won't cause pandemic
- Mood debuff is short and minor and needed more to attract attention of consumer that they ate something bad if they didn't examine the food
- When they eat it, they have a 90% chance to NOT get sick
- Cleaning is very easy. You can get a space cleaner bottle and just sanitize any food/pulls with one click

Darkshadowrogue commented:
And for others I made a few improvements in scope of this PR:
> 
>     * Food left on griddle will no longer decompose, so you can leave your eggs and steaks there safely now
> 
>     * When you craft food and your hands are full, it is automatically placed on nearby tables

Yeah you could just make the improvements, mate. I'm not going to address the rest of the things in your comment - I don't see why I need to. It's all pretty common sense, rewriting what you've already written in a different way doesn't make them better, it's still the same, and for those same things I think it's pretty obvious why the things you've said aren't beneficial to anybody, IC or OC.

Exester509 commented:
> @Darkshadowrogue I do play chef. I play it like @Roryl-c above, respecting food hygiene for no real reason. This doesn't add any extra overhead in terms of tedium. All you need is just to not drop stuff on the floor deliberately. Many chefs already avoid this for IC reasons. And for others I made a few improvements in scope of this PR:
> 
> * Food left on griddle will no longer decompose, so you can leave your eggs and steaks there safely now
> * When you craft food and your hands are full, it is automatically placed on nearby tables
> 
> Ants are not really punishing because of very long delay. You need the food to really stay on floor for a long time for ants to spawn, so it is barely a punishment. And I am already making adjustments that should address many concerns in comments above, so please keep in mind the whole picture.
> 
> * It will be picked from a list of diseases that will be unique, have static medicine, non-transmissive, so that it won't cause pandemic
> * Mood debuff is short and minor and needed more to attract attention of consumer that they ate something bad if they didn't examine the food
> * When they eat it, they have a 90% chance to NOT get sick
> * Cleaning is very easy. You can get a space cleaner bottle and just sanitize any food/pulls with one click

On a sidenote, if you're going for realism, I wouldn't make it space cleaner dependant.
Spraying cleaning products in your food is a TERRIBLE idea

KathrinBailey commented:
![image](https://github.com/tgstation/tgstation/assets/53862927/c39aebac-83c2-4a60-9dee-749a695a45cc)

So what's stopping me from doing this roundstart?

Roryl-c commented:
> ![image](https://user-images.githubusercontent.com/53862927/246655340-c39aebac-83c2-4a60-9dee-749a695a45cc.png)
> 
> So what's stopping me from doing this roundstart?

Not like there's a shortage of ways to grief roundstart. What's stopping you from frying all of it or stealing all of it or burning all of it? Throwing it on the floor is the only one trivially fixed by washing.

MTandi committed:
gastritium, carpellosis

KathrinBailey commented:
> Not like there's a shortage of ways to grief roundstart. What's stopping you from frying all of it or stealing all of it or burning all of it? Throwing it on the floor is the only one trivially fixed by washing.

Well I'd imagine the chef with CQC is there to stop such things roundstart.

Roryl-c commented:
> So what's stopping me from doing this roundstart?

> Well I'd imagine the chef with CQC is there to stop such things roundstart.

There's your answer then I guess 😄

MTandi committed:
nebula nausea

MTandi committed:
Merge branch 'master' into floor_food

MTandi committed:
typo

MTandi committed:
Merge branch 'master' into floor_food

MTandi committed:
emissive nebula

MTandi committed:
all kitchen machines are elevated

MTandi committed:
emissive fix

MTandi committed:
gibber elevated too

MTandi committed:
early qdel issue

MTandi committed:
better max stage check

MTandi committed:
Merge branch 'master' into floor_food

ChungusGamer666 commented on code/datums/components/food/germ_sensitive.dm:
```diff
@@ -0,0 +1,124 @@
+/datum/component/germ_sensitive/proc/wash()
+	if(infective)
+		infective = FALSE
+		qdel(parent.GetComponent(/datum/component/infective))
```
just 
qdel(src)

MTandi replied:
Why? It won't become dirty again then.

ChungusGamer666 replied:
my bad i read this wrong, thought it was nuking the germ sensitive component

ChungusGamer666 replied:
you can resolve this

ChungusGamer666 commented:
is it joever

MTandi committed:
Merge branch 'master' into floor_food
